scripts/download_data.py

```diff
@@ -0,0 +1,256 @@
+"""
+Constructs a data source for the ga4gh server by downloading data from
+authoritative remote servers.
+"""
+# TODO
+# - need some kind of checkpoint functionality to resume process
+#   where it left off since getting a clean run is so rare...
+# - references support
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import glob
+import gzip
+import os
+import tempfile
+import urllib2
+
+import pysam
+
+import utils
+
+
+def mkdirAndChdirList(dirNameList):
+    """
+    For each entry in dirNameList, make a directory if it does not exist and
+    cd into it
+    """
+    for directory in dirNameList:
+        mkdirAndChdir(directory)
+
+
+def mkdirAndChdir(dirName):
+    """
+    Make a directory if it does not exist and cd into it
+    """
+    if not os.path.exists(dirName):
+        os.mkdir(dirName)
+    os.chdir(dirName)
+
+
+def cleanDir():
+    """
+    Removes genomic files in current directory
+    """
+    cwd = os.getcwd()
+    utils.log("Cleaning out directory '{}'".format(cwd))
+    globs = [
+        "*.tbi", "*.vcf", "*.vcf.gz", "*.bam", "*.bam.bai", "*.fasta.gz",
+        "*.fasta", "*.fasta.fai"]
+    for fileGlob in globs:
+        fileNames = glob.glob(fileGlob)
+        for fileName in fileNames:
+            os.remove(fileName)
+
+
+def escapeDir(levels=4):
+    # back to orig dir
+    for _ in range(levels):
+        os.chdir('..')
+
+
+class AbstractFileDownloader(object):
+    """
+    Base class for individual site genome file downloaders
+    """
+    def __init__(self, args):
+        self.args = args
+        self.maxPos = 0
+        self.minPos = 2**30
+        self.datasetId = 'dataset1'
+        self.variantSetId = self.args.source
+        self.referenceSetId = 'main'
+
+    def _getVcfFilenames(self):
+        baseFileName = (
+            "ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5a"
+            ".20130502.genotypes.vcf.gz")
+        chrNames = [str(i) for i in range(1, 23)]
+        fileNames = [baseFileName.format(chrName) for chrName in chrNames]
+        # the X and Y files use a different filename prefix
+        fileNames.append(
+            'ALL.chrX.phase3_shapeit2_mvncall_integrated_v1a'
+            '.20130502.genotypes.vcf.gz')
+        fileNames.append(
+            'ALL.chrY.phase3_integrated_v1a.20130502.genotypes.vcf.gz')
+        return fileNames
+
+    def getVcfBaseUrl(self):
+        return self.getBaseUrl() + '/ftp/release/20130502/'
```

Comment:
Use ``os.path.join`` here -- this is currently giving a ``//`` in the middle of the URL.